sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -1044,22 +1047,31 @@ private void messageReceived(final ChannelHandlerContext context, final RntbdRes
                     switch (subStatusCode) {
                         case SubStatusCodes.COMPLETING_SPLIT_OR_MERGE:
                             cause = new PartitionKeyRangeIsSplittingException(error, lsn, partitionKeyRangeId, responseHeaders);
+                            handleGoneException(serviceRequest, cause);
+                            rntbdConnectionStateListener.attemptBackgroundAddressRefresh(serviceRequest);
                             break;
                         case SubStatusCodes.COMPLETING_PARTITION_MIGRATION:
                             cause = new PartitionIsMigratingException(error, lsn, partitionKeyRangeId, responseHeaders);
+                            handleGoneException(serviceRequest, cause);
+                            rntbdConnectionStateListener.attemptBackgroundAddressRefresh(serviceRequest);
                             break;
                         case SubStatusCodes.NAME_CACHE_IS_STALE:
                             cause = new InvalidPartitionException(error, lsn, partitionKeyRangeId, responseHeaders);
+                            handleGoneException(serviceRequest, cause);
+                            rntbdConnectionStateListener.attemptBackgroundAddressRefresh(serviceRequest);
                             break;
                         case SubStatusCodes.PARTITION_KEY_RANGE_GONE:
                             cause = new PartitionKeyRangeGoneException(error, lsn, partitionKeyRangeId, responseHeaders);
+                            rntbdConnectionStateListener.attemptBackgroundAddressRefresh(serviceRequest);
```

Comment:
why are we not handlingGoneException in thsis case?

Reply:
I wanted to understand how to tweak request-specific flags for 410 / 1002 scenarios since `PartitionKeyRangeGoneRetryPolicy` doesn't set anything explicitly. We can take this offline.